rs/bitcoin/ckbtc/minter/src/tasks.rs

```diff
@@ -27,14 +34,9 @@ pub struct TaskQueue {
     deadline_by_task: BTreeMap<TaskType, u64>,
 }
 
-fn set_global_timer(ts: u64) {
+fn set_global_timer<R: CanisterRuntime>(ts: u64, runtime: &R) {
     LAST_GLOBAL_TIMER.with(|v| v.set(ts));
-
-    // SAFETY: setting the global timer is always safe; it does not
-    // mutate any canister memory.
-    unsafe {
```

Comment:
I wonder why `unsafe` was used here? Your new implementation `IcCanisterRuntime` does not seem to need it.

Reply:
it's 🪄 . I basically asked the Rust cdk team at some point to provide an API method for `ic0::global_timer_set`, which corresponds to `ic_cdk::api::set_global_timer`, meaning that for previous versions of the CDK this method was not there and one needed to use `unsafe` (the comment is wrong, it's about the fact that this method is only available if the code runs on an IC replica)